pkg/routing/adapter/filter/reconciler.go

```diff
@@ -67,8 +68,12 @@ func (r *Reconciler) reconcile(ctx context.Context, f *v1alpha1.Filter) error {
 	return nil
 }
 
-// ObserveFinalizeKind implements reconcilerv1alpha1.ReadOnlyFinalizer.
-func (r *Reconciler) ObserveFinalizeKind(ctx context.Context, f *v1alpha1.Filter) reconciler.Event {
+// ObserveDeletion implements reconciler.OnDeletionInterface.
+func (r *Reconciler) ObserveDeletion(ctx context.Context, key types.NamespacedName) error {
+	f := &v1alpha1.Filter{}
+	f.SetName(key.Name)
+	f.SetNamespace(key.Namespace)
+
 	return r.finalize(ctx, f)
```

Comment:
@tzununbekov question not related to the validity of this PR.

Are calls from this reconciler to `RegisterHandlerFor` and `DeregisterHandlerFor` noop, and thus can be removed and there would be no need to implement `OnDeletionInterface`?

Reply:
Hm, nice catch. Not sure why we even bothered implementing a dummy interface.
https://github.com/triggermesh/triggermesh/blob/54e355dc6990364225b38f41fdf1390aa13f4423/pkg/routing/adapter/filter/adapter.go#L96-L104

Reply:
right, I think I copied these dummy methods from somewhere else where they were implemented.

Reply:
See #49